scalding-core/src/main/scala/com/twitter/scalding/ExecutionUtil.scala

```diff
@@ -2,17 +2,27 @@ package com.twitter.scalding
 
 import com.twitter.algebird.Semigroup
 
+import scala.util.{Failure, Success, Try}
+
 object ExecutionUtil {
   /**
+   * Useful for when using parallelism and want to output the final state of the job
+   */
+  def liftedSequenceUnit[T](executions: Execution[Seq[Try[T]]], handleComplete: Try[T] => Unit): Execution[Unit] =
+    executions.onComplete {
+      case Success(sequence) => sequence.foreach(handleComplete)
+      case Failure(ex) => () //If the execution was lifted this should never occur
+    }.unit
+
+  /**
    * Generate a list of executions from a date range
    *
    * @param duration Duration to split daterange
-   * @param parallelism How many jobs to run in parallel
    * @param fn Function to run a execution given a date range
    * @return Sequence of Executions per Day
    */
-  def executionsFromDates[T](duration: Duration, parallelism: Int = 1)(fn: DateRange => Execution[T])(implicit dr: DateRange): Seq[Execution[T]] =
-    dr.each(duration).map(fn).toSeq
+  def executionsFromDates[T](duration: Duration)(fn: DateRange => Execution[T])(implicit dr: DateRange): Seq[Execution[(DateRange, T)]] =
```

Comment:
you can't change signatures like this in general. This breaks both source and binary compatibility. If you want the date inside, return in with fn. Seems not worth the change.

Dropping parallelism seems good since it was unused.

Maybe this should be private. Is is really helpful? If we are making incompatibilities, I'd rather be conservative and drop APIs we are not sure of.